R/palaeorotate.R

```diff
@@ -211,60 +173,53 @@ palaeorotate <- function(occdf, lng = "lng", lat = "lat", age = "age",
                          uncertainty = TRUE, round = 3) {
   # Error-handling ----------------------------------------------------------
   if (!exists("occdf") || !is.data.frame(occdf)) {
-    stop("Please supply occdf as a dataframe")
+    stop("Please supply `occdf` as a data.frame.")
   }
 
   if (any(c(lng, lat, age) %in% colnames(occdf) == FALSE)) {
-    stop("defined `lng`, `lat`, or `age` not found in `occdf`")
+    stop("Defined `lng`, `lat`, or `age` not found in `occdf`.")
   }
 
   if (any(!is.numeric(occdf[, lat]), is.na(occdf[, lat]),
           !is.numeric(occdf[, lng]), is.na(occdf[, lng]),
           !is.numeric(occdf[, age]), is.na(occdf[, age]))) {
-    stop("`lng`, `lat` and `age` should be of class numeric")
+    stop("`lng`, `lat` and `age` should be of class numeric.")
   }
 
   if (any(occdf[, age] < 0)) {
     stop("`age` contains negative values. Input ages should be positive.")
   }
 
   if (sum(abs(occdf[, lat]) > 90) != 0) {
-    stop("`lat` values should be >= -90\u00B0 and <= 90\u00B0")
+    stop("`lat` values should be >= -90\u00B0 and <= 90\u00B0.")
   }
 
   if (sum(abs(occdf[, lng]) > 180) != 0) {
-    stop("`lng` values should be >= -180\u00B0 and <= 180\u00B0")
+    stop("`lng` values should be >= -180\u00B0 and <= 180\u00B0.")
   }
 
   if (method %in% c("grid", "point") == FALSE) {
-    stop("`method` should be grid or point")
+    stop("`method` should be either 'grid' or 'point'.")
   }
 
   if (!is.null(round) && !is.numeric(round)) {
-    stop("`round` should be NULL or of class numeric")
+    stop("`round` should be NULL or of class numeric.")
   }
 
   if (!is.logical(uncertainty)) {
-    stop("`uncertainty` should be of class logical (TRUE/FALSE)")
+    stop("`uncertainty` should be of class logical (TRUE/FALSE).")
   }
 
-  # Add warnings for use of mantle reference frame models
-  if (any(model %in% c("MULLER2022", "MULLER2019",
-                   "MATTHEWS2016_mantle_ref"))) {
-    warning(paste0("Selected model(s) use a mantle reference frame and are ",
-    "not recommended for reconstructing palaeocoordinates. See details."))
+  # Add stop for removed models
+  if (any(model %in% c("MULLER2022", "MULLER2019", "MULLER2016",
+                       "MATTHEWS2016_mantle_ref", "SETON2012"))) {
+    stop(paste0("Selected model(s) have recently been removed as they are not",
+    " in a palaeomagnetic reference frame. See details for available models."))
```

Comment:
I would maybe suggest displaying the problematic models. One possibility:
```
deprecated_models <- c("MULLER2022", "MULLER2019", "MATTHEWS2016_mantle_ref", "SETON2012")
  if (any(model %in% deprecated_models)) {
paste0("Selected model(s) have recently been removed as they are not",
       " in a palaeomagnetic reference frame: ", 
       paste(deprecated_models[which(model %in% deprecated_models)], collapse = ", "),
       ". See details for available models.")
}
```

Reply:
Yeah, that's fair enough! Thanks!